dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterParams.java

```diff
@@ -46,4 +49,22 @@
 
     @Builder.Default
     private final Set<String> filters = Sets.newHashSet( "*" );
+
+    public static <O> FieldFilterParams<O> of( List<O> objects, List<String> filters )
+    {
+        return FieldFilterParams
+            .<O> builder()
+            .objects( objects )
+            .filters( Sets.newHashSet( StringUtils.join( filters, "," ) ) )
```

Comment:
This is confusing - we join a list of filters with comma and make it into a set with one member? Hm..
could use `singletonSet` but still odd API.

Reply:
Yeah, this is mainly to make it work ok with spring mvc, for example `?fields=id,name,groups[id,name,code]` in spring mvc will end up being a List of `id` `name` `groups[id` `name` `code]` which means in a set it will actually remove name (since its double. It's a hack for now, maybe we can discuss on tomorrows platform call.

Reply:
Previously I have picked it up from the `HttpServletRequest` but was hoping to not have to do it.. anyways all this will be hidden behind something like `@FieldFilter(...) FieldFilterParams params` in the future

Reply:
I have changed it to use `Collections.singleton(...)`